src/main/java/com/benko/exercise/sort/BubbleSorter.java

```diff
@@ -0,0 +1,31 @@
+package com.benko.exercise.sort;
+
+public class BubbleSorter {
+
+    public void sort(int[] numbers) {
+        for (int i = 0; i < numbers.length; i++) {
+            swapElements(numbers);
+        }
+    }
+
+    private static void swapElements(int[] numbers) {
+        for (int i = 0; i < numbers.length-1; i++) {
+            if (isGreaterThan(i, i+1, numbers)) {
+                int temp;
+                temp = numbers[i + 1];
+                numbers[i + 1] = numbers[i];
+                numbers[i] = temp;
+            }
+        }
+    }
+
+    private static boolean isGreaterThan(int x, int y, int[] numbers) {
+        return numbers[x]>numbers[y];
+    }
+
+    public static void printNumbers(int[] sortedNumbers) {
+        for (int i = 0; i < sortedNumbers.length; i++) {
```

Comment:
idea suggests to use a for-each loop here. the syntax is very easy:
```suggestion
        for (int n : sortedNumbers) {
            System.out.println(n);
        }
```

Reply:
done

Reply:
that's better but... why `x`?
by convention, variables in such iterations have names corresponding to the type of things in the collection/array.
And there can be two forms: full and short.
Examples of the full:
```
for (int number : numbers)
for (String name : names)
for (Item item : items)
for (User user : users)
```
Short:
```
for (int n : numbers)
for (String n : names)
for (Item i : items)
for (User user : users)
```

So you see that it was not accident that I suggested variable name `n`.

Reply:
ah I see, ok make sense.
And the x from the first website under 'for-each java' Google query:
https://javastart.pl/baza-wiedzy/darmowy-kurs-java/podstawy-jezyka/petla-for-each

Reply:
now i see the next weakness of free courses...

Reply:
Done